### PR TITLE
chore(version) Remove const_format dependency

### LINKED ISSUE #13496: Remove `const_format` dependency
### Describe the feature

Its only usage is when formatting version specifiers, e.g. https://github.com/paradigmxyz/reth/blob/de477ba91a95e0e339f1b111788818e37de22800/crates/node/core/src/version.rs#L19

We can move this logic into the build script which makes it more readable and removes the dependency.

This practically means replacing the const_format calls with `env!("NAME")`, and moving the formatting logic to build.rs with `println!("cargo:rustc-env=NAME={value}")`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by duvbell**
Closes #13496

### CODE REVIEW

**Comment by duvbell** [comment]
hello @DaniPopes , can you help me review this PR ?

**Review by mattsse — COMMENTED** [review]
this looks good,
let's also remove the const_format dep now?

**Comment by DaniPopes** [comment]
few notes
- BUILD_PROFILE_NAME can be replaced by emitting env::var("PROFILE");
- LONG_VERSION has multiple lines which you have to emit individually, like `for (i, line) in .lines().enumerate() { /*emit "LONG_VERSION{i}"*/ }`

**Comment by duvbell** [comment]
@DaniPopes tks, i updated the code regarding your comment

File: crates/node/core/build.rs [hunk 1]
```diff
@@ -22,10 +22,68 @@ fn main() -> Result<(), Box<dyn Error>> {
     // if not on a tag: v0.2.0-beta.3-82-g1939939b
     // if on a tag: v0.2.0-beta.3
     let not_on_tag = env::var("VERGEN_GIT_DESCRIBE")?.ends_with(&format!("-g{sha_short}"));
-    let is_dev = is_dirty || not_on_tag;
-    println!("cargo:rustc-env=RETH_VERSION_SUFFIX={}", if is_dev { "-dev" } else { "" });
+    let version_suffix = if is_dirty || not_on_tag { "-dev" } else { "" };
```
Comment (mattsse) [inline crates/node/core/build.rs:25]: unrelated, I noticed that some docker builds have a -dev suffix, which doesn't make sense, so I think this is kinda wrong

wouldn't be opposed to removing this entirely @onbjerg

**Review by mattsse — COMMENTED** [review]
pending @DaniPopes

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**